opencti-platform/opencti-front/src/components/dataGrid/DataTableHeader.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { FunctionComponent, useContext } from 'react';
+import { ArrowDropDown, ArrowDropUp, MoreVert } from '@mui/icons-material';
+import IconButton from '@mui/material/IconButton';
+import SimpleDraggrable from 'react-draggable';
+import makeStyles from '@mui/styles/makeStyles';
+import { createStyles } from '@mui/styles';
+import { Theme as MuiTheme } from '@mui/material/styles/createTheme';
+import { DataTableContext } from './dataTableUtils';
+import { DataTableColumn, DataTableHeaderProps, LocalStorageColumns } from './dataTableTypes';
+import { isNotEmptyField } from '../../utils/utils';
+
+export const SELECT_COLUMN_SIZE = 42;
+
+const useStyles = makeStyles<MuiTheme, { column: DataTableColumn }>((theme) => createStyles({
+  headerContainer: {
+    position: 'relative',
+    display: 'flex',
+    width: ({ column }) => `calc(var(--header-${column?.id}-size) * 1px)`,
+    fontWeight: 'bold',
+    justifyContent: 'center',
+    alignItems: 'center',
+    height: 40,
+    '&:hover': {
+      '& $draggable': {
+        background: theme.palette.secondary.main,
+      },
+    },
+  },
+  headerAligner: {
+    paddingLeft: 8,
+    display: 'flex',
+    alignItems: 'center',
+    whiteSpace: 'nowrap',
+    overflow: 'hidden',
+    cursor: ({ column: { isSortable } }) => (isSortable ? 'pointer' : 'unset'),
+  },
+  aligner: { flexGrow: 1 },
+  draggable: {
+    position: 'absolute',
+    top: 0,
+    right: 3,
+    height: '100%',
+    width: 4,
+    borderRadius: 2,
+    cursor: 'col-resize',
+    userSelect: 'none',
+    touchAction: 'none',
+    zIndex: 999,
+  },
+}));
+
+const DataTableHeader: FunctionComponent<DataTableHeaderProps> = ({
+  column,
+  setAnchorEl,
+  setActiveColumn,
+  setLocalStorageColumns,
+  containerRef,
+  sortBy,
+  orderAsc,
+}) => {
+  const classes = useStyles({ column });
+
+  const {
+    columns,
+    setColumns,
+    availableFilterKeys,
+    onSort,
+  } = useContext(DataTableContext);
+
+  return (
+    <div
+      key={column.id}
+      className={classes.headerContainer}
+    >
+      <div
+        className={classes.headerAligner}
+        onClick={(e) => {
+          // Small debounce
+          (e.target as HTMLDivElement).style.setProperty('pointer-events', 'none');
+          setTimeout(() => {
+            (e.target as HTMLDivElement).style.setProperty('pointer-events', 'auto');
+          }, 800);
+          if (column.isSortable) {
+            onSort!(column.id, !orderAsc);
+          }
+        }}
+      >
+        {column.label}
+        {sortBy && (orderAsc ? <ArrowDropUp /> : <ArrowDropDown />)}
+      </div>
+      <>
+        <div className={classes.aligner} />
+        {(column.isSortable || isNotEmptyField(availableFilterKeys)) && (
+          <>
+            <IconButton
+              onClick={(e) => {
```

Comment:
```suggestion
            <IconButton
              style={{ marginRight: 10 }}
              onClick={(e) => {
```

adding some margins to prevent the draggable bar to be on top of the icon. This was making the interaction a bit sensitive.